src/Components/Facility/AddBedForm.tsx

```diff
@@ -125,7 +127,7 @@ export const AddBedForm = (props: BedFormProps) => {
       name,
       description,
       bed_type: bedType,
-      number_of_beds: numberOfBeds,
+      number_of_beds: headerText === "Update Bed" ? 1 : numberOfBeds,
```

Comment:
Avoid depending on visual texts when possible.

```suggestion
      number_of_beds: bedId ? 1 : numberOfBeds,
```